app/models/concerns/pdf_visualization_operations.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+module PdfVisualizationOperations
+  extend ActiveSupport::Concern
+
+  FORM_IDENTIFIER_PATTERN = /([^\/]+)\/(\d+\.\d+)\z/
+
+  included do
+    def prepare_pdf_transformation
+      return unless upvs_form&.xsl_fo
+      return unless unsigned_content
+      return unless xml?
+
+      template = Nokogiri::XSLT(upvs_form.xsl_fo)
+      fo_xml = template.transform(xml_unsigned_content)
+      fo_xml.encoding = 'UTF-8'
+
+      begin
+        xml_file = Tempfile.new("#{id}.xml")
+        xml_file.write(xml_unsigned_content)
+        xml_file.rewind
+
+        xsl_file = Tempfile.new("#{id}.fo.xsl")
+        xsl_file.write(upvs_form.xsl_fo.to_s)
+        xsl_file.rewind
+
+        pdf_file = Tempfile.new("#{id}.pdf")
+
+        system "fop -xml #{xml_file.path} -c #{Rails.root + 'config/fop.xconf'} -xsl #{xsl_file.path} -pdf #{pdf_file.path}"
```

Comment:
Ake rychle je toto? Nevadi, ze to budeme volat synchronne?

Reply:
Zda sa mi to byt v pohode, skusime si my cez feature flag a mozeme prehodnotit